unlock-app/src/components/interface/Recipts/elements/ReceiptBox.tsx

```diff
@@ -0,0 +1,297 @@
+import { Button, Disclosure } from '@unlock-protocol/ui'
+import ReactToPrint from 'react-to-print'
+import { useRef, useState } from 'react'
+import { PoweredByUnlock } from '../../checkout/PoweredByUnlock'
+import { addressMinify } from '~/utils/strings'
+import { UpdatePurchaserDrawer } from './UpdatePurchaserDrawer'
+import { useUpdateReceipt, useGetReceipt } from '~/hooks/receipts'
+import dayjs from 'dayjs'
+import networks from '@unlock-protocol/networks'
+import { useAuth } from '~/contexts/AuthenticationContext'
+import { useLockManager } from '~/hooks/useLockManager'
+import { useWeb3Service } from '~/utils/withWeb3Service'
+import { useQuery } from '@tanstack/react-query'
+import Link from 'next/link'
+import { useGetPrice } from '~/hooks/usePrice'
+
+interface ReceiptBoxProps {
+  lockAddress: string
+  network: number
+  hash: string
+}
+
+interface DetailLabelProps {
+  label: string
+  value: string
+  inline?: boolean
+}
+
+export const DetailLabel = ({
+  label,
+  value,
+  inline = false,
+}: DetailLabelProps) => {
+  return (
+    <div className={`flex ${inline ? 'gap-2' : 'flex-col'}`}>
+      <span className="text-base">{label}</span>
+      <div className="flex items-center">
+        <span className="block text-base font-bold break-words md:inline-block">
+          {value || '-'}
+        </span>
+      </div>
+    </div>
+  )
+}
```

Comment:
Think it's been repeated a few times in similar form. Maybe open a PR to separate it into a reusable and consistent component.

Reply:
ok! ill open an issue for it!